### PR TITLE
Team 5 - Map Page

### PR DESCRIPTION
**Pull request by AlexRobinson01**
A map page reachable by the bottom tab bar, powered by MapsUI and OpenStreetMap. It can be zoomed into/out of and dragged around, like a normal map, and requires very few modifications to the existing codebase.

### CODE REVIEW

File: Map/MapPage.xaml.cs [hunk 1]
```diff
@@ -3,13 +3,47 @@
...
+ */
 public partial class MapPage : ContentPage
 {
 	public MapPage()
 	{
 		InitializeComponent();
 
         MapControl mapControl = new();
+        // Initialize the map with OpenStreetMap's API and add it to MapPage's content
         mapControl.Map?.Layers.Add(OpenStreetMap.CreateTileLayer());
```
Comment (blocksombr31) [inline Map/MapPage.xaml.cs:47]: I'm glad one of us was actually able to find a free map to use!

**Comment by rogersm-uwosh** [comment]
I can't get this to compile on iOS, some error involving the SkiaSharp.dll not being found. I will need to defer approving this PR until we can get it to work on both platforms (that's the whole point of cross-platform). Team 4 (Carissa, Hannah and Pachia) were able to get this to work with the standard Map component built into .NET MAUI, could you reach to them to figure out how they did it?

**Comment by AlexRobinson01** [comment]
> I can't get this to compile on iOS, some error involving the SkiaSharp.dll not being found. I will need to defer approving this PR until we can get it to work on both platforms (that's the whole point of cross-platform). Team 4 (Carissa, Hannah and Pachia) were able to get this to work with the standard Map component built into .NET MAUI, could you reach to them to figure out how they did it?

I looked this error up and it seems like it might be a certificate problem on Mac where the project is built rather than anything related to SkiaSharp... thanks, Apple. I'm not sure if that's something you'd want to try fixing on your end, but I don't think any of us can from here since it seems to be device-specific when building the project.

As for the Map control provided by Microsoft.Maui.Controls.Maps, according to their documentation this is not supported on Windows due to a missing WinUI control, and the package they redirect to (CommunityToolkit.Maui.Maps) is not working any more due to Bing no longer giving out new API keys for their maps. That was why we went with MapsUI, every other option was either deprecated/missing entirely on some platforms or they needed API keys which require entering credit card info even if they don't need payment.

Team 4 has an API key in their PR which is evidence of this, so unless you'd prefer that key is kept in the repo where we can also use it, we're sort of running thin on options.

**Comment by rogersm-uwosh** [comment]
This is now working, thank you for your contribution to the app!

**Pull request merged**